### PR TITLE
fix: og images avatars

### PR DESCRIPTION
**Pull request by zomars**
## What does this PR do?

Fixes some edge cases for OG avatars not displaying properly

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

- Verify generated images are displaying properly in public pages

## Mandatory Tasks

- [ ] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

### CODE REVIEW

File: apps/web/modules/users/views/users-public-view.getServerSideProps.tsx [hunk 1]
```diff
@@ -206,7 +207,10 @@ export const getServerSideProps: GetServerSideProps<UserPageProps> = async (cont
 
   const profile = {
     name: user.name || user.username || "",
-    image: user.avatar,
+    image: getUserAvatarUrl({
+      ...user,
+      profile: user.profile,
+    }),
```
Comment (zomars) [inline apps/web/modules/users/views/users-public-view.getServerSideProps.tsx:213]: This was returning `"image": "/peer/avatar.png"` which is wrong.

image

File: packages/lib/constants.ts [hunk 2]
```diff
@@ -32,7 +32,11 @@ export const SENDER_NAME = process.env.NEXT_PUBLIC_SENDGRID_SENDER_NAME || "Cal.
 // This is the URL from which all Cal Links and their assets are served.
 // Use website URL to make links shorter(cal.com and not app.cal.com)
 // As website isn't setup for preview environments, use the webapp url instead
-export const CAL_URL = new URL(WEBAPP_URL).hostname.endsWith(".vercel.app") ? WEBAPP_URL : WEBSITE_URL;
+// If it's a .vercel.app domain, keep it.
+// Else use the website url if defined and finally fallback to the webapp url
+export const CAL_URL = new URL(WEBAPP_URL).hostname.endsWith(".vercel.app")
+  ? WEBAPP_URL
+  : process.env.NEXT_PUBLIC_WEBSITE_URL || WEBAPP_URL;
```
Comment (zomars) [inline packages/lib/constants.ts:39]: Avoid hard-coding `cal.com` as much as possible. In our hosted instance NEXT_PUBLIC_WEBSITE_URL is always set as `https://cal.com` so there should not be any problems here. Also this won't disturb self-hosts.

File: packages/lib/getAvatarUrl.ts [hunk 3]
```diff
@@ -1,11 +1,11 @@
...
+import { AVATAR_FALLBACK, CAL_URL, WEBAPP_URL } from "@calcom/lib/constants";
+import type { Team, User } from "@calcom/prisma/client";
 import type { UserProfile } from "@calcom/types/UserProfile";
 
 /**
  * Gives an organization aware avatar url for a user
  * It ensures that the wrong avatar isn't fetched by ensuring that organizationId is always passed
+ * It should always return a fully formed url
  */
 export const getUserAvatarUrl = (
```
Comment (zomars) [inline packages/lib/getAvatarUrl.ts:10]: Refactored so we always return fully formed URLs

File: packages/lib/OgImages.tsx [hunk 4]
```diff
@@ -62,7 +62,7 @@ export const constructMeetingImage = (
     `?type=meeting`,
     `&title=${encodeURIComponent(title)}`,
     `&meetingProfileName=${encodeURIComponent(profile.name)}`,
-    profile.image && `&meetingImage=${encodeURIComponent(WEBAPP_URL + profile.image)}`,
```
Comment (zomars) [inline packages/lib/OgImages.tsx:65]: image should return fully formed always.

File: packages/lib/OgImages.tsx [hunk 5]
```diff
@@ -125,7 +125,7 @@ export const Meeting = ({ title, users = [], profile }: MeetingImageProps) => {
   const avatars = attendees
     .map((user) => {
       if ("image" in user && user?.image) return user.image;
-      if ("username" in user && user?.username) return `${WEBAPP_URL}/${user.username}/avatar.png`;
+      if ("username" in user && user?.username) return `${CAL_URL}/${user.username}/avatar.png`;
```
Comment (zomars) [inline packages/lib/OgImages.tsx:128]: This is a last resort fallback.

**Review by zomars — COMMENTED** [review]
Self review done

**Review by PeerRich — DISMISSED** [review]

**Review by keithwillcode — DISMISSED** [review]

**Comment by keithwillcode** [comment]
Code looks good. Need to test

**Review by emrysal — COMMENTED** [review]
More inconsistencies! nice spot 😮‍💨

**Review by emrysal — APPROVED** [review]

**Pull request merged**